module/userProjects/helper.js

```diff
@@ -2987,6 +3152,119 @@ module.exports = class UserProjectsHelper {
+					throw {
+						status: HTTP_STATUS_CODE.bad_request.status,
+						message: CONSTANTS.apiResponses.USER_PROJECT_NOT_FOUND,
+					}
+				}
+
+				// Get the first element of project details array
+				projectDetails = projectDetails[0]
+
+				// Check if the project is not submitted or certificate is not eligible
+				if (
+					projectDetails.status !== CONSTANTS.common.SUBMITTED_STATUS ||
+					!projectDetails.certificate ||
+					!projectDetails.certificate.eligible
+				) {
+					throw {
+						status: HTTP_STATUS_CODE.bad_request.status,
+						message: CONSTANTS.apiResponses.PROJECT_NOT_ELIGIBLE_FOR_CERTIFICATE,
+					}
+				}
+
+				// Check if transaction ID is missing or empty
+				if (!projectDetails.certificate.transactionId || projectDetails.certificate.transactionId === '') {
+					throw {
+						status: HTTP_STATUS_CODE.bad_request.status,
+						message: CONSTANTS.apiResponses.CERTIFICATE_NOT_AVAILABLE_FOR_THE_PROJECT,
+					}
+				}
+				// Check if both PDF path and SVG path are missing or empty
+				else if (
+					!(projectDetails.certificate.pdfPath && projectDetails.certificate.pdfPath != '') &&
+					!(projectDetails.certificate.svgPath && projectDetails.certificate.svgPath != '')
+				) {
+					throw {
+						status: HTTP_STATUS_CODE.bad_request.status,
+						message: CONSTANTS.apiResponses.CERTIFICATE_NOT_AVAILABLE_FOR_THE_PROJECT,
+					}
+				}
+
+				let pdfCertificateUrl, svgCertificateUrl
+				const certificateUrls = await cloudServicesHelper.getDownloadableUrl([
+					projectDetails.certificate.pdfPath ? projectDetails.certificate.pdfPath : '',
+					projectDetails.certificate.svgPath ? projectDetails.certificate.svgPath : '',
+				])
+				if (certificateUrls && certificateUrls.result && certificateUrls.result.length > 0) {
+					for (const certificateUrlsIndex of certificateUrls.result) {
+						if (
+							projectDetails.certificate.pdfPath != '' &&
+							certificateUrlsIndex.payload['sourcePath'] != '' &&
+							certificateUrlsIndex.payload['sourcePath'] == projectDetails.certificate.pdfPath
+						) {
+							pdfCertificateUrl = certificateUrlsIndex.url
+						} else if (
+							projectDetails.certificate.svgPath != '' &&
+							certificateUrlsIndex.payload['sourcePath'] != '' &&
+							certificateUrlsIndex.payload['sourcePath'] == projectDetails.certificate.svgPath
+						) {
+							svgCertificateUrl = certificateUrlsIndex.url
+						}
+					}
+				}
+
+				// Prepare certificate verification information
+				const certificateVerificationInfo = {
+					projectId,
+					projectName: projectDetails.title,
+					programId: projectDetails.programId,
+					solutionId: projectDetails.solutionId,
+					userId: projectDetails.userId,
+					isCertificateVerified: true,
+					completedDate: projectDetails.completedDate,
+					issuedOn: projectDetails.issuedOn,
+					eligible: projectDetails.certificate.eligible,
+					pdfCertificateUrl: pdfCertificateUrl ? pdfCertificateUrl : '',
```

Comment:
@Prajwal17Tunerlabs keep as certificatePdfUrl and certificateSvgUrl

Reply:
variables name changed to suggested names.